src/app/Dashboard/AddCard.tsx

```diff
@@ -157,3 +233,124 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
     </>
   );
 };
+
+interface PropsConfigFormProps {
+  cardTitle: string;
+  controls: PropControl[];
+  initialState: any;
+  onChange: ({}) => void;
+}
+
+const PropsConfigForm = (props: PropsConfigFormProps) => {
+  const [propsConfig, setPropsConfig] = React.useState(props.initialState);
```

Comment:
Looks like we are having selected propConfig as a state here and its parent.

I think we could pass in a prop as `selectedPropConfig` in `PropsConfigFormProps` from parent and remove this state. This way, we can reduce re-rendering and limit "source of truth"?